Blish HUD/Controls/Textbox.cs

```diff
@@ -1,233 +1,136 @@
-        public Color ForeColor {
-            get => _foreColor;
-            set => SetProperty(ref _foreColor, value);
-        }
+        private int _prevCursorIndex  = 0;
+        private int _horizontalOffset = 0;
 
-        private TimeSpan _lastInvalidate;
-        private bool _textWasChanged = false;
+        public TextBox() {
+            _multiline = false;
+            _maxLength = 2048;
 
-        protected bool _caretVisible = false;
-        private bool CaretVisible {
-            get => _caretVisible;
-            set => SetProperty(ref _caretVisible, value);
+            this.Size = new Point(STANDARD_CONTROLWIDTH, STANDARD_CONTROLHEIGHT);
         }
 
-        protected BitmapFont _font = Content.DefaultFont14;
-        public BitmapFont Font {
-            get => _font;
-            set => SetProperty(ref _font, value);
+        protected override void HandleEnter() {
+            OnEnterPressed(EventArgs.Empty);
         }
 
-        /// <summary>
-        /// [NOT THREAD-SAFE]
-        /// </summary>
-        public TextBox() {
-            _lastInvalidate = DateTime.MinValue.TimeOfDay;
-
-            _mttb = new System.Windows.Forms.TextBox() {
-                Parent                   = BlishHud.Form,
-                Size                     = new Size(20, 20),
-                Location                 = new System.Drawing.Point(-500),
-                AutoCompleteMode         = AutoCompleteMode.Append,
-                AutoCompleteSource       = AutoCompleteSource.CustomSource,
-                AutoCompleteCustomSource = new AutoCompleteStringCollection(),
-                ShortcutsEnabled         = true,
-                TabStop                  = false
-            };
-
-            _sharedInterceptor.LeftMouseButtonReleased += delegate {
-                if (_sharedInterceptor.ActiveControl == this) {
-                    Textbox_LeftMouseButtonReleased(null, null);
-                }
-            };
+        protected override void MoveLine(int delta) { /* NOOP */ }
 
-            _mttb.TextChanged += InternalTextBox_TextChanged;
-            _mttb.KeyDown     += InternalTextBox_KeyDown;
-            _mttb.KeyUp       += InternalTextBox_KeyUp;
+        protected override void OnClick(MouseEventArgs e) {
+            base.OnClick(e);
 
-            this.Size = Standard.Size;
+            int newIndex = GetCursorIndexFromFromPosition(this.RelativeMousePosition.X - TEXT_HORIZONTALPADDING);
 
-            Input.Mouse.LeftMouseButtonPressed  += Input_MouseButtonPressed;
-            Input.Mouse.RightMouseButtonPressed += Input_MouseButtonPressed;
-        }
-
-        protected override void OnMouseEntered(MouseEventArgs e) {
-            bool restoreFocus = _mttb.Focused;
-
-            _sharedInterceptor.Show(this);
-
-            if (restoreFocus) {
-                Textbox_LeftMouseButtonReleased(null, null);
+            if (_cursorIndex == newIndex && e.IsDoubleClick) {
+                SelectAll();
```

Comment:
Shouldn't this select the word instead of the hole text?

Reply:
The answer to this is _yes_, but I don't think it will be included in this PR.  Should just be a matter of searching in both directions for the word boundary and updating the selection and would probably be a method on `TextInputBase` since both would use it so I'll make note to have it added in the future.

Reply:
Just kidding.  I went ahead and got it implemented.  🙂